segmentation/utils/synthetic_spatial_datagen.py

```diff
@@ -222,10 +219,142 @@ def generate_test_label_map(size_img=(1024, 1024), num_A=100, num_B=100, num_C=1
+
+    # generate an array of numbers valued depending on how close or far away the coordinate is
+    # from the center or the membrane
+    # note that we multiply by -random() for the membrane case because the values
+    # will initially be negative and it will help to make the values consistent
+    # also, we need to center the distribution around the center for nuclear
+    # but around an edge value for membrane
+    if expression == "nuclear":
+        prob_mask = size_img[0] + size_img[1] - np.maximum(np.abs(prob_mask_rows - cell_center_x),
+                                                           np.abs(prob_mask_cols - cell_center_y)) * random()
+        center_value = prob_mask[cell_center_x, cell_center_y]
+        prob_mask = norm.pdf(prob_mask, center_value)
+    else:
+        prob_mask = np.maximum(np.abs(prob_mask_rows - cell_center_x),
+                               np.abs(prob_mask_cols - cell_center_y)) - (size_img[0] + size_img[1]) * -random()
+        edge_value = prob_mask[cell_center_x, cell_center_y - radius]
+        prob_mask = norm.pdf(prob_mask, edge_value)
+
+    # extract only the coordinates defined by the cell region and set the prob mask accordingly
+    signal_data[cell_region[0], cell_region[1]] = prob_mask[cell_region[0], cell_region[1]]
+
+    return signal_data
+
+
+def generate_two_cell_test_regions(size_img, radius):
+    """
+    This function generates the region locations of each cell in our test segmentation channel mask.
+    We will assume we'll be generating circles as cells.
+
+    We will assume we have only two cells for the time being, one cell gets assigned marker 1 and the other
+    marker 2
+
+    Args:
+        size_img: the dimensions of the image
+        radius: the radius of each cell
+
+    Returns:
+        region_coverage: a dictionary mapping each cell marker to its respective coordinates
+            which will be used to cover said marker (a tuple in the format (x_coords, y_coords))
+        region_centers: a dictionary mapping each cell marker to its respective center (also a tuple format)
+    """
+
+    # test that the radius specified is within the boundaries
+    # to prevent any overflow, we ask that the radius be no larger than
+    # half of any dimension passed
+    if radius > size_img[0] / 2 or radius > size_img[1] / 2:
+        raise ValueError("Radius specified is larger than one of the image dimensions")
+
+    # generate the two cells at the top left of the image
+    center_1 = (radius, radius)
+    center_2 = (radius, radius * 3 + 1)
+
+    # draw the coordnates covered for the two cell
+    x_coords_cell_1, y_coords_cell_1 = circle(center_1[0], center_1[1], radius + 1)
+    x_coords_cell_2, y_coords_cell_2 = circle(center_2[0], center_2[1], radius + 1)
+
+    # now store the regions generated by skimage.draw.circle into a dict with the key
+    # identifying the marker
+    cell_regions = {1: (x_coords_cell_1, y_coords_cell_1),
```

Comment:
I think instead of storing the actual coordinates of the cells, we should just store the value we assigned them in a segmentation label mask, and then pass the mask.